File: .snippets/code/smart-contracts/libraries/web3-js/fetchLastBlock.js
const { Web3 } = require('web3');

const createProvider = (rpcUrl) => {
  const web3 = new Web3(rpcUrl);
  return web3;
};

const PROVIDER_RPC = {
  rpc: 'https://testnet-passet-hub-eth-rpc.polkadot.io',
  chainId: 420420422,
  name: 'polkadot-hub-testnet',
};

const main = async () => {
  try {
    const web3 = createProvider(PROVIDER_RPC.rpc);
    const latestBlock = await web3.eth.getBlockNumber();
    console.log('Last block: ' + latestBlock);
  } catch (error) {
    console.error('Error connecting to Polkadot Hub TestNet: ' + error.message);
  }
};

main();

File: .snippets/code/smart-contracts/libraries/web3-py/fetch_last_block.py
from web3 import Web3


def main():
    try:
        PROVIDER_RPC = "https://services.polkadothub-rpc.com/testnet"
        web3 = Web3(Web3.HTTPProvider(PROVIDER_RPC))
        latest_block = web3.eth.block_number
        print("Last block: " + str(latest_block))
    except Exception as error:
        print("Error connecting to Polkadot Hub TestNet: " + str(error))


if __name__ == "__main__":
    main()
